mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/job/IMantisJobMetadata.java

```diff
@@ -194,4 +194,6 @@ public interface IMantisJobMetadata {
      * is past timeout secs
      */
     long getWorkerTimeoutSecs();
+
+    Optional<String> getStageAttribute(int stageNum, String sizeName);
```

Comment:
not sure i understand what this function does with stageNum + sizeName. Can you add an example to doc?

Reply:
my bad this function is supposed to return the value of a given key of stage container attributes (which may be null or the key may not exist). let me rename `sizeName` with something more appropriate and rename the method to reflect this. 

The idea here is to extract the size name from the stage info containerAttributes map.